Absolute and norm tests fail

**Describe the bug**
```
=================================== FAILURES ===================================
___________________________ test_quaternion_absolute ___________________________
test/test_quaternion.py:585: in test_quaternion_absolute
    assert q.abs() == a
E   assert 6.024948132556828 == 6.024948132556827
E     -6.024948132556828
E     +6.024948132556827
_____________________________ test_quaternion_norm _____________________________
test/test_quaternion.py:599: in test_quaternion_norm
    assert q.norm() == a
E   assert 36.300000000000004 == 36.3
E     -36.300000000000004
E     +36.3
=================== 2 failed, 66 passed, 4 xfailed in 15.05s ===================
```


**To Reproduce**
Build the package in Arch:
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=python-numpy-quaternion

**Expected behavior**
Pass test

**Environment (please complete the following information):**
 - Arch `Linux 5.3.1-zen1-1-zen `
 - Arch system packages and AUR
 - Numpy version `1.17.2`
```
numpy.__config__.show()                                                                                         
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
  NOT AVAILABLE
atlas_3_10_blas_threads_info:
  NOT AVAILABLE
atlas_3_10_blas_info:
  NOT AVAILABLE
atlas_blas_threads_info:
  NOT AVAILABLE
atlas_blas_info:
  NOT AVAILABLE
accelerate_info:
  NOT AVAILABLE
blas_info:
    libraries = ['cblas', 'blas', 'blas']
    library_dirs = ['/usr/lib64']
    include_dirs = ['/usr/local/include', '/usr/include']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
blas_opt_info:
    define_macros = [('NO_ATLAS_INFO', 1), ('HAVE_CBLAS', None)]
    libraries = ['cblas', 'blas', 'blas']
    library_dirs = ['/usr/lib64']
    include_dirs = ['/usr/local/include', '/usr/include']
    language = c
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
  NOT AVAILABLE
openblas_clapack_info:
  NOT AVAILABLE
flame_info:
  NOT AVAILABLE
atlas_3_10_threads_info:
  NOT AVAILABLE
atlas_3_10_info:
  NOT AVAILABLE
atlas_threads_info:
  NOT AVAILABLE
atlas_info:
  NOT AVAILABLE
lapack_info:
    libraries = ['lapack', 'lapack']
    library_dirs = ['/usr/lib64']
    language = f77
lapack_opt_info:
    libraries = ['lapack', 'lapack', 'cblas', 'blas', 'blas']
    library_dirs = ['/usr/lib64']
    language = c
    define_macros = [('NO_ATLAS_INFO', 1), ('HAVE_CBLAS', None)]
    include_dirs = ['/usr/local/include', '/usr/include']
```

 - quaternion `2019.7.23.15.26.49`

**Additional context**
Add any other context about the problem here.

--- a/test/test_quaternion.py
+++ b/test/test_quaternion.py
@@ -425,6 +425,7 @@ def test_as_spherical_coords(Rs):
                      for i in range(5000)]
     for vartheta, varphi in random_angles:
         vartheta2, varphi2 = quaternion.as_spherical_coords(quaternion.from_spherical_coords(vartheta, varphi))
+        varphi2 = (varphi2 + 2*np.pi) if varphi2 < 0 else varphi2
         assert abs(vartheta - vartheta2) < 1e-12, ((vartheta, varphi), (vartheta2, varphi2))
         assert abs(varphi - varphi2) < 1e-12, ((vartheta, varphi), (vartheta2, varphi2))
     # Now test that arbitrary rotors rotate z to the appropriate location
@@ -582,7 +583,7 @@ def test_quaternion_absolute(Qs):
     for q, a in [(Qs[q_0], 0.0), (Qs[q_1], 1.0), (Qs[x], 1.0), (Qs[y], 1.0), (Qs[z], 1.0),
                  (Qs[Q], np.sqrt(Qs[Q].w ** 2 + Qs[Q].x ** 2 + Qs[Q].y ** 2 + Qs[Q].z ** 2)),
                  (Qs[Qbar], np.sqrt(Qs[Q].w ** 2 + Qs[Q].x ** 2 + Qs[Q].y ** 2 + Qs[Q].z ** 2))]:
-        assert q.abs() == a
+        assert np.allclose(q.abs(), a)
 
 
 def test_quaternion_norm(Qs):
@@ -596,7 +597,7 @@ def test_quaternion_norm(Qs):
     for q, a in [(Qs[q_0], 0.0), (Qs[q_1], 1.0), (Qs[x], 1.0), (Qs[y], 1.0), (Qs[z], 1.0),
                  (Qs[Q], Qs[Q].w ** 2 + Qs[Q].x ** 2 + Qs[Q].y ** 2 + Qs[Q].z ** 2),
                  (Qs[Qbar], Qs[Q].w ** 2 + Qs[Q].x ** 2 + Qs[Q].y ** 2 + Qs[Q].z ** 2)]:
-        assert q.norm() == a
+        assert np.allclose(q.norm(), a)
 
 
 # Unary quaternion returners

